zap/zap.go

```diff
@@ -14,89 +15,173 @@ var (
 	_ slog.Logger = (*Logger)(nil)
 )
 
-// LoggerEntry is either a zap.Logger or a zap.SugaredLogger
-type LoggerEntry interface{}
-
 // Logger is an adaptor using go.uber.org/zap as slog.Logger
 type Logger struct {
-	entry LoggerEntry
+	logger *zap.Logger
+	cfg    *zap.Config
 }
 
 // Enabled tells this logger is enabled
-func (zl *Logger) Enabled() bool {
+func (zpl *Logger) Enabled() bool {
+	if zpl == nil || zpl.logger == nil || zapcore.LevelOf(zpl.logger.Core()) == zapcore.InvalidLevel {
+		return false
+	}
+
 	return true
 }
 
 // WithEnabled passes the logger and if it's enabled
-func (zl *Logger) WithEnabled() (slog.Logger, bool) {
-	return zl, true
+func (zpl *Logger) WithEnabled() (slog.Logger, bool) {
+	return zpl, zpl.Enabled()
 }
 
 // Print adds a log entry with arguments handled in the manner of fmt.Print
-func (zl *Logger) Print(args ...any) {
-	zl.print(fmt.Sprint(args...))
+func (zpl *Logger) Print(args ...any) {
+	zpl.print(fmt.Sprint(args...))
 }
 
 // Println adds a log entry with arguments handled in the manner of fmt.Println
-func (zl *Logger) Println(args ...any) {
-	zl.print(fmt.Sprintln(args...))
+func (zpl *Logger) Println(args ...any) {
+	zpl.print(fmt.Sprintln(args...))
 }
 
 // Printf adds a log entry with arguments handled in the manner of fmt.Printf
-func (zl *Logger) Printf(format string, args ...any) {
-	zl.print(fmt.Sprintf(format, args...))
+func (zpl *Logger) Printf(format string, args ...any) {
+	zpl.print(fmt.Sprintf(format, args...))
```

Comment:
if `Enabled()` can use `zpl.logger.Check()` we could save the string formatting cost by testing before calling `print`

Reply:
Optimized to use Enabled()